src/bundles/analytics.test.js

```diff
@@ -27,152 +27,169 @@ function createStore (analyticsOpts = {}) {
       selectDesktopCountlyActions: () => ([])
     },
     createAnalyticsBundle(analyticsOpts)
-  )()
+  )({
+    // tests using mockAnalyticsCachedState are mimicking the createCacheBundle that sets initial state
+    analytics: mockAnalyticsCachedState
+  })
 }
 
-it('should disable analytics by default', () => {
-  const store = createStore()
-  expect(store.selectAnalyticsEnabled()).toBe(false)
-  // no events will be sent as no consents have been given
-  expect(store.selectAnalyticsConsent()).toEqual([])
-  // user has not explicitly opted in or out yet
-  expect(global.Countly.opt_in).not.toHaveBeenCalled()
-  expect(global.Countly.opt_out).not.toHaveBeenCalled()
+it('should render', () => {})
+describe('new/returning user default behavior', () => {
+  it('should enable analytics by default for new user who has not opted in or out', () => {
+    const store = createStore()
+    expect(global.Countly.opt_in).toHaveBeenCalled()
+    expect(global.Countly.opt_in.mock.calls.length).toBe(1)
+    // events will be sent as consents have been given by default
+    expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views', 'location'])
+  })
+  it('should enable existing analytics by default for returning user who was opted_in', () => {
+    const mockDefaultState = {
+      lastEnabledAt: Date.now(),
+      lastDisabledAt: 0,
+      showAnalyticsBanner: false,
+      optedOut: false,
+      consent: ['sessions', 'events', 'views']
+    }
+    const store = createStore({}, mockDefaultState)
+    expect(store.selectAnalyticsEnabled()).toBe(true)
+    expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views'])
+    // should not show analytics banner for these users
+    expect(store.selectShowAnalyticsBanner()).toBe(false)
+  })
+  it('should enable analytics for returning user who opted_out prior to new opt-in by default updates', () => {
+    const mockDefaultState = {
+      lastEnabledAt: 0,
+      lastDisabledAt: Date.now(),
+      showAnalyticsBanner: false,
+      optedOut: false,
+      consent: []
+    }
+    const store = createStore({}, mockDefaultState)
+    expect(global.Countly.opt_in).toHaveBeenCalled()
+    expect(global.Countly.opt_in.mock.calls.length).toBe(1)
+    expect(store.selectAnalyticsConsent()).toEqual(['sessions', 'events', 'views', 'location'])
+    // should show analytics banner for these users
+    expect(store.selectShowAnalyticsBanner()).toBe(true)
+  })
```

Comment:
nit: technically, the name here should be "...prior to new opt-out requirement" because the policy of telemetry is "opt-out" which means users are implicitly granting permissions. The action they have to perform to change the default is "opt-out". 

The name test is :+1: though.